contracts/BC_fusion/StakePool.sol

```diff
@@ -0,0 +1,266 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-upgradeable/utils/structs/DoubleEndedQueueUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20PermitUpgradeable.sol";
+
+import "./System.sol";
+
+interface IStakeHub {
+    function unbondPeriod() external view returns (uint256);
+    function transferGasLimit() external view returns (uint256);
+}
+
+contract StakePool is Initializable, ReentrancyGuard, System, ERC20PermitUpgradeable {
+    using DoubleEndedQueueUpgradeable for DoubleEndedQueueUpgradeable.Bytes32Deque;
+
+    /*----------------- constant -----------------*/
+    uint256 public constant MAX_CLAIM_NUMBER = 20;
```

Comment:
I think we dont have to limit in on contract side, the frontend will construct a proper vaule.

Reply:
Removed